hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -302,36 +304,59 @@ private boolean getFilesForArchive(DBCheckpoint checkpoint,
       logEstimatedTarballSize(checkpoint, includeSnapshotData);
     }
 
-    // Get the active fs files.
-    Path dir = checkpoint.getCheckpointLocation();
-    if (!processDir(dir, copyFiles, hardLinkFiles, sstFilesToExclude,
-        new HashSet<>(), excluded, copySize, null)) {
-      return false;
-    }
 
-    if (!includeSnapshotData) {
-      return true;
-    }
+    if (includeSnapshotData) {
+      // Get the snapshot files.
+      Set<Path> snapshotPaths = getSnapshotDirs(checkpoint, true);
+      Path snapshotDir = getSnapshotDir();
+      boolean processedSnapshotData = processDir(snapshotDir, copyFiles, hardLinkFiles,
+          sstFilesToExclude, snapshotPaths, excluded, copySize, null);
+      if (LOG.isDebugEnabled()) {
+        if (processedSnapshotData) {
+          LOG.debug("Finished processing OM snapshot data");
+        }
+      }
+      if (!processedSnapshotData) {
+        return false;
+      }
 
-    // Get the snapshot files.
-    Set<Path> snapshotPaths = getSnapshotDirs(checkpoint, true);
-    Path snapshotDir = getSnapshotDir();
-    if (!processDir(snapshotDir, copyFiles, hardLinkFiles, sstFilesToExclude,
-        snapshotPaths, excluded, copySize, null)) {
-      return false;
-    }
-    // Process the tmp sst compaction dir.
-    if (!processDir(sstBackupDir.getTmpDir().toPath(), copyFiles, hardLinkFiles,
-        sstFilesToExclude, new HashSet<>(), excluded, copySize,
-        sstBackupDir.getOriginalDir().toPath())) {
-      return false;
+      // Process the tmp sst compaction dir.
+      boolean processedTmpSstCompactionDir = processDir(sstBackupDir.getTmpDir().toPath(),
+          copyFiles, hardLinkFiles, sstFilesToExclude, new HashSet<>(),
+              excluded, copySize, sstBackupDir.getOriginalDir().toPath());
+      if (LOG.isDebugEnabled()) {
+        if (processedTmpSstCompactionDir) {
+          LOG.debug("Finished processing tmp SST Compaction Dir");
+        }
+      }
+      if (!processedTmpSstCompactionDir) {
+        return false;
+      }
+
+      // Process the tmp compaction log dir.
+      boolean processedTmpCompactionLogDir = processDir(compactionLogDir.getTmpDir().toPath(),
+          copyFiles, hardLinkFiles, sstFilesToExclude, new HashSet<>(),
+              excluded, copySize, compactionLogDir.getOriginalDir().toPath());
```

Comment:
nit: let's just call it `processedCompactionLogDir`. We create tmp for local referencing only.
```suggestion
      boolean processedCompactionLogDir = processDir(compactionLogDir.getTmpDir().toPath(),
          copyFiles, hardLinkFiles, sstFilesToExclude, new HashSet<>(),
              excluded, copySize, compactionLogDir.getOriginalDir().toPath());
```